File: src/csharp/Problems/ReverseList.cs
//https://leetcode.com/problems/reverse-linked-list/
namespace LeetCode.Problems;

public sealed class ReverseList : ProblemBase
{
    public static void Run()
    {
        var d = Run(new ListNode(1, new ListNode(2, new ListNode(3, new ListNode(4, new ListNode(5))))));
    }

    private static ListNode Run(ListNode head)
    {
        var result = head;
        var current = result.next;
        do
        {
            var next = current.next;

            current.next = result;
            result = current;
            current = next;
        }
        while (current != null);

        return result;
    }
}

File: src/python/problems/ReverseList.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class ReverseList:
    def Solution(self, head: Optional[ListNode]) -> Optional[ListNode]:
        result = None
        current = head
        while current:
            next = current.next
            current.next = result
            result = current
            current = next

        return result


ReverseList().Solution(ListNode(1, ListNode(2, ListNode(3))))
